sdk/metric/version_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"github.com/stretchr/testify/assert"
+	"regexp"
+	"testing"
+)
+
+// regex taken from https://github.com/Masterminds/semver/tree/v3.1.1
+var versionRegex = regexp.MustCompile(`^v?([0-9]+)(\.[0-9]+)?(\.[0-9]+)?` +
```

Comment:
NIT: I think this regex is a bit loose for what we are expecting for our version numbers, e.g. `v` and minor and patch numbers won't be optional.

Reply:
@MadVikingGod i came across a[ previous discussion](https://github.com/open-telemetry/opentelemetry-go/pull/2156#discussion_r690387427) for the same issue . Should i replace this with your [suggestion](https://ihateregex.io/expr/semver/)? Does that mean this needs to change in all places in code? for example [this](https://github.com/open-telemetry/opentelemetry-go/blob/d5fca833d6f6fc75e092c2108e0265aa778b8923/version_test.go#L27)

Reply:
SemVer does NOT allow `v` prefix.

Reply:
@pellared i have changed regex string as per [official recommendation ](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)